src/Simplex/Chat/Store/Groups.hs

```diff
@@ -1195,7 +1229,16 @@ deleteSentProbe db userId probeId =
     (userId, probeId)
 
 matchReceivedProbe :: DB.Connection -> User -> Contact -> Probe -> IO (Maybe Contact)
-matchReceivedProbe db user@User {userId} _from@Contact {contactId} (Probe probe) = do
+matchReceivedProbe db user@User {userId} _from@(Contact {contactId}) p@(Probe probe) =
```

Comment:
Instead of making it a single function, it would probably be more flexible to move matchReceivedProbe_ get query into a separate function "getMatchingProbe" and reuse it in matchReceivedProbe and matchReceivedMemberProbe, with duplicating these couple of lines:
```
case contactIds of
    [] -> pure Nothing
    cId : _ -> eitherToMaybe <$> runExceptT (getContact db user cId)
```